File: leetcode/0206-reverse-linked-list/solution.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun reverseList(head: ListNode?): ListNode? {
        if(head == null) {
            return head
        }
        var ret = head
        var node = ret?.next
        ret?.next = null
        while(node != null) {
            val new_node = node?.next
            node?.next = ret
            ret = node
            node = new_node
        }
        return ret
    }
}

File: leetcode/0206-reverse-linked-list/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
    if head == nil {
        return head
    }
    node := head.Next
    head.Next = nil
    for node != nil {
        new_node := node.Next
        node.Next = head
        head = node
        node = new_node
    }
    return head
}
